internal/testutils/stats/test_metrics_recorder.go

```diff
@@ -36,18 +35,16 @@ import (
 // have taken place. It also persists metrics data keyed on the metrics
 // descriptor.
 type TestMetricsRecorder struct {
-	t *testing.T
-
 	intCountCh   *testutils.Channel
 	floatCountCh *testutils.Channel
 	intHistoCh   *testutils.Channel
 	floatHistoCh *testutils.Channel
 	intGaugeCh   *testutils.Channel
 
-	// Mu protects Data.
+	// Mu protects data.
 	Mu sync.Mutex
```

Comment:
The main goal of making a getter was so that `mu` didn't need to be exported and to make access simpler, since mutexes are generally internal state and complicate things.

Seems like you need something more complex if you need an atomic read of multiple values (though this wasn't supported previously?).  `func Metrics(name []string) (*float64)`???

Reply:
Hmmmmm I just deleted the grab of it in balancer_test. I was worried due to the low time for weight updates (when time.Ticker goes off and reupdates scheduler) but I made it 30 seconds so should never race (it'll fail much earlier than that). So mutex still grabbed sometimes in this component in exported helpers.